### DISCUSSION TITLE
Rond Rows Filter

### DISCUSSION THREAD

**Discussion by SimoneBergonzi** [mia-platform/community#267] [Q&A] [answered]
### Which section?

Console

### Which is the version?

v9.5.3

### Description

Hi,

I'm trying to configure some Rows Filter policies within the RBAC section of the console but I can't get it work.

The filter rule is the following:

```
approved_subscriptions_policy {
    resource := data.resources[_]
    resource.status == "APPROVED"
}
``` 

The idea is to keep only the documents with status: "APPROVED".

The rbac-service is logging the following, resulting in a 403:
```
crud-service-6697568969-scjvk > rbac-service {
  "allowed": false,
  "evaluationTimeMicroseconds": 180,
  "level": 20,
  "matchedPath": "/subscriptions/",
  "method": "GET",
  "msg": "policy evaluation completed",
  "partialEval": false,
  "policyName": "approved_subscriptions_policy",
  "reqId": "73e692946a108e5ebb702f6b434efa03",
  "requestedPath": "/subscriptions/",
  "time": 1686228832050
}
crud-service-6697568969-scjvk > rbac-service {
  "level": 50,
  "msg": "policy resulted in not allowed",
  "policyName": "approved_subscriptions_policy",
  "reqId": "73e692946a108e5ebb702f6b434efa03",
  "time": 1686228832050
}
crud-service-6697568969-scjvk > rbac-service {
  "error": {
    "message": "RBAC policy evaluation failed, user is not allowed",
    "policyName": "approved_subscriptions_policy"
  },
  "level": 50,
  "msg": "RBAC policy evaluation failed",
  "reqId": "73e692946a108e5ebb702f6b434efa03",
  "time": 1686228832050
}
```

I don't understand.
With that row filter, I was expecting RBAC to add a filter in the header "acl_rows", so at worst, I should get an empty list, but it seems that RBAC is interpreting the row filter as an Allow Policy returning **403**.

Is there any configuration I should check on my crud-service (or somewhere else) in order to use rows filters with RBAC?

**Additional infos**
Allow Policies seems to work properly, for example, the following is working:
```
company_admin_policy {
    input.request.method == "POST"
    input.request.body.role == "ADMIN"
}
```

In addition, Rows Filter seems to work fine on a different crud-service.
The following rule is configured on a service which forwards the request to the crud-service of another project and it works just fine:
```
mobility_users_policy {
    input.request.method == "GET"
    resource := data.resources[_]
    resource.groups == "mobility"
}
```

producing this log on the crud-service:
```
{
  "level": 20,
  ...
  "query": {
    "$and": [
      {
        "$and": [
          {
            "$or": [
              {
                "$and": [
                  {
                    "groups": {
                      "$eq": "mobility"
                    }
                  }
                ]
              }
            ]
          }
        ]
      },
      {
        "__STATE__": "PUBLIC"
      }
    ]
  },
  "projection": [
   ...
  ],
  "options": {},
  "msg": "findAll operation requested"
}
```

**Comment by fredmaggiowski** [comment, marked as answer]
Hi, did you configure the policy to be used as a "query generation policy"? 

Are you using the Manual Routes section or the OAS in your service? Can you share your configuration (a screenshot of the manual route or the OAS segment should be enough) ?

**Reply by SimoneBergonzi** [reply]
Yep, that was the problem!
I hadn't set the toggle `Apply row filter on request` when configuring the policy in the `Manual route` section.

I solved the problem following the instruction of a collegue, I have to admint I couldn't find it in the documentation...

![Screenshot from 2023-06-08 15-45-47](https://github.com/mia-platform/community/assets/26777944/771f6196-52eb-4bc3-8090-e4cf61d0a935)

**Reply by fredmaggiowski** [reply]
great!
